server/sonar-server/src/main/java/org/sonar/server/issue/ws/BulkChangeAction.java

```diff
@@ -109,17 +111,20 @@
   private final IssueStorage issueStorage;
   private final NotificationManager notificationService;
   private final List<Action> actions;
-  private final IssueChangeTrigger issueChangeTrigger;
+  private final QGChangeEventFactory qgChangeEventFactory;
+  private final QGChangeEventListeners qgChangeEventListeners;
 
-  public BulkChangeAction(System2 system2, UserSession userSession, DbClient dbClient, IssueStorage issueStorage, NotificationManager notificationService, List<Action> actions,
-    IssueChangeTrigger issueChangeTrigger) {
+  public BulkChangeAction(System2 system2, UserSession userSession, DbClient dbClient, IssueStorage issueStorage,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Constructor has 8 parameters, which is greater than 7 authorized. ([squid:S00107](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsl%2FSONAR-10085%2Fupgrade_qg_change_event_api_with_issue_details&issues=AWDCAGAZqDrCo5ysgzHB&open=AWDCAGAZqDrCo5ysgzHB)

[ISSUE_KEY]: AWDCAGAZqDrCo5ysgzHB